tests/drivers/hwinfo/api/testcase.yaml

```diff
@@ -0,0 +1,3 @@
+tests:
+  peripheral.device_id:
+    tags: driver
```

Comment:
You can add another line `depends_on: hwinfo` here, and then in the appropriate boards add `hwinfo` to the supported list. I think that should allow you to remove the #ifdef that @galak asked about in line 28.

Reply:
@MaureenHelm since this is a feature of the SoC, is there a possibility to add hwinfo to SoC instead of boards?